components/script/dom/htmlanchorelement.rs

```diff
@@ -581,21 +582,39 @@ pub fn follow_hyperlink(subject: &Element, hyperlink_suffix: Option<String>) {
     let source = document.browsing_context().unwrap();
 
     // Step 4-5: target attribute.
-    let target_attribute_value = subject.get_attribute(&ns!(), &local_name!("target"));
-
+    let subject_type = subject.get_attribute(&ns!(), &local_name!("target"));
+
+    let mut target_attribute_value = None;
+    if subject.is::<HTMLAreaElement>() || subject.is::<HTMLAnchorElement>() {
+        target_attribute_value = subject_type;
+
+        if !subject.has_attribute(&local_name!("target")) {
+            let base = (*document).base_element();
+            match base {
+                Some(base) => {
+                    let base_html_unwrap = (*base).upcast::<HTMLElement>().upcast::<Element>();
+                    if base_html_unwrap.has_attribute(&local_name!("target")) {
+                        target_attribute_value =
+                            base_html_unwrap.get_attribute(&ns!(), &local_name!("target"));
+                    } else {
+                        target_attribute_value = None;
+                    }
+                },
+                None => target_attribute_value = None,
+            };
+        }
+    }
     // Step 6.
     let noopener = if let Some(link_types) = subject.get_attribute(&ns!(), &local_name!("rel")) {
         let values = link_types.Value();
-        let contains_noopener = values.contains("noopener");
-        let contains_noreferrer = values.contains("noreferrer");
-        let contains_opener = values.contains("opener");
         let target_is_blank = if let Some(name) = target_attribute_value.as_ref() {
             name.Value().to_lowercase() == "_blank"
         } else {
             false
         };
-
-        contains_noreferrer || contains_noopener || (!contains_opener && target_is_blank)
+        values.contains("noreferrer") ||
+            values.contains("noopener") ||
+            (!values.contains("opener") && target_is_blank)
     } else {
         false
     };
```

Comment:
This would also be good to extract into a function that links to https://html.spec.whatwg.org/multipage/links.html#get-an-element's-noopener and splits the string value appropriately (https://html.spec.whatwg.org/multipage/links.html#linkTypes)